### PR TITLE
Add circuit template how-to topic

### PR DESCRIPTION
**Pull request by shannynalayna**
Adds an app-dev guide using the `circuit-template` feature in Splinter

### CODE REVIEW

File: docs/0.5/howto/using_circuit_templates.md [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+  https://creativecommons.org/licenses/by/4.0/
+-->
+
+This guide explains how to use the `circuit-template` feature of Splinter,
+including explaining the concept and format of a circuit template file and
+how to use the `circuit-template` feature when developing an application.
+
+## Overview
+
+A circuit template contains a set of `rule`s which partially define a new circuit.
```
Comment (peterschwarz) [inline docs/0.5/howto/using_circuit_templates.md:15]: I don't think that rule needs to be in code-quotes.

File: docs/0.5/howto/using_circuit_templates.md [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+    set-metadata:
+        encoding: json
+        metadata:
+            - key: "scabbard_admin_keys"
+              value: ["$(ADMIN_KEYS)"]
+            - key: "alias"
+              value: "$(GAMEROOM_NAME)"
+```
+
+There are three main sections to the circuit template, `version`, `arg`s, and
```
Comment (peterschwarz) [inline docs/0.5/howto/using_circuit_templates.md:60]: These don't match what's in the example: they should be `version`, `args`, and `rules`.  The same is true throughout the rest of the docs on these sections.

File: docs/0.5/howto/using_circuit_templates.md [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+  required: false
+  default: $(SIGNER_PUB_KEY)
+  description: >-
+    Public keys used to verify transactions in the scabbard service
+```
+
+The value of `SIGNER_PUB_KEY` is used as the default value of the `ADMIN_KEYS`
+argument.
+
+## Rules
```
Comment (peterschwarz) [inline docs/0.5/howto/using_circuit_templates.md:96]: In this section, I would remove much of the code quotes, except when directly referring to the `rules` section.  Each individual thing is a "rule", but it is not a code term that anyone uses when constructing a template.

File: docs/0.5/howto/using_circuit_templates.md [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+`GAMEROOM_NAME` is `gameroom_name` or `ADMIN_KEYS` is `admin_keys`. The current
+circuit template `rule`s implementations can be found in the
+`circuit::template::rules` module.
+
+## Environment Variables
+
+Circuit template storage directories can be specified using the
+`SPLINTER_CIRCUIT_TEMPLATE_PATH` environment variable. Multiple storage
+directories may be specified in this environment variable, paths delineated by
+‘:’. Using multiple directories does have an effect on the behavior of the
```
Comment (peterschwarz) [inline docs/0.5/howto/using_circuit_templates.md:143]: Was the intent to make this a code-quoted `:`?

**Review by peterschwarz — CHANGES_REQUESTED** [review]

File: docs/0.5/howto/using_circuit_templates.md [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+
+A circuit template contains a set of `rule`s which partially define a new circuit.
+This can make circuit creation substantially easier for users as only a minimal
+amount of information needs be provided--the remainder is defined in the circuit
+template.
+
+The following is an example circuit template YAML file specific to the Gameroom
+application:
+
+```
```
Comment (vaporos) [inline docs/0.5/howto/using_circuit_templates.md:23]: I think if you do ```yaml here you will see syntax highlighting.

File: docs/0.5/howto/using_circuit_templates.md [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+            - key: "alias"
+              value: "$(GAMEROOM_NAME)"
+```
+
+There are three main sections to the circuit template, `version`, `arg`s, and
+`rule`s.
+
+## Version
+
+This describes the version of circuit the template is targeting.
```
Comment (vaporos) [inline docs/0.5/howto/using_circuit_templates.md:65]: This is the version of the template file, not the version of the circuit.

**Review by peterschwarz — COMMENTED** [review]
Is this document intended to enumerate the available rules?  Or is that in a separate document?

**Review by peterschwarz — APPROVED** [review]

**Review by vaporos — APPROVED** [review]

**Pull request merged**